pkg/pipeline/sink/m3u8/writer.go

```diff
@@ -0,0 +1,75 @@
+package m3u8
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type PlaylistWriter struct {
+	filename       string
+	targetDuration int
+}
+
+func NewPlaylistWriter(filename string, targetDuration int) (*PlaylistWriter, error) {
+	p := &PlaylistWriter{
+		filename:       filename,
+		targetDuration: targetDuration,
+	}
+
+	f, err := os.Create(p.filename)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	var sb strings.Builder
+	sb.WriteString("#EXTM3U\n")
+	sb.WriteString("#EXT-X-VERSION:4\n")
+	sb.WriteString("#EXT-X-PLAYLIST-TYPE:EVENT\n")
+	sb.WriteString("#EXT-X-ALLOW-CACHE:NO\n")
+	sb.WriteString("#EXT-X-MEDIA-SEQUENCE:0\n")
+	sb.WriteString(fmt.Sprintf("#EXT-X-TARGETDURATION:%d\n", p.targetDuration))
+
+	_, err = f.WriteString(sb.String())
+	if err != nil {
+		return nil, err
+	}
+
+	return p, nil
+}
+
+func (p *PlaylistWriter) Append(dateTime time.Time, duration float64, filename string) error {
+	f, err := os.OpenFile(p.filename, os.O_WRONLY|os.O_APPEND, fs.ModeAppend)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	var sb strings.Builder
+	sb.WriteString("#EXT-X-PROGRAM-DATE-TIME:")
+	sb.WriteString(dateTime.UTC().Format(time.RFC3339Nano))
```

Comment:
The spec mentions SO/IEC 8601:2004 as the spec for the date which is almost the same, but with millisecond accuracy instead of nanosecond. Using nanosecond may break some parsers.